tests/foreman/ui/test_computeresource_azurerm.py

```diff
@@ -236,9 +226,7 @@ def test_positive_azurerm_host_provision_ud(
                 assert azurecloud_vm.type == AZURERM_VM_SIZE_DEFAULT
 
         finally:
-            azure_vm = session_puppet_enabled_sat.api.Host().search(
-                query={'search': f'name={fqdn}'}
-            )
+            azure_vm = sat_azure.api.Host().search(query={'search': f'name={fqdn}'})
             if azure_vm:
                 with satellite_setting('destroy_vm_on_host_delete=True'):
```

Comment:
Since this test is parametrized to both `module_target_sat` and `session_puppet_enabled_sat`, how do you ensure that the setting is updated on `session_puppet_enabled_sat`, not on `module_target_sat`?
I see that `satellite_setting` is still importing `entities`.

Reply:
Great catch @ogajduse !

Reply:
What I am thinking here is with the parametrization. However, the `satellite_setting` is still importing entities, I believe it works in an orderly fashion of execution, and also if we look closely it's trying to delete the VM on that particular satellite instance if it doesn't get the VM the scenario will automatically break. 
```

with satellite_setting('destroy_vm_on_host_delete=True'):
                azure_vm[0].delete(synchronous=False)
```

or I could just delete the host with the way I am searching it. @jyejare @ogajduse @Gauravtalreja1

Reply:
Addressed